modules/polls/polls.js

```diff
@@ -0,0 +1,251 @@
+        if (!poll || !choices || !question) {
+            return;
+        }
+
+        if (this.poll && this.choices && this.question) {
+            logger.warn('Attempt to create a poll '
+                + 'while one is currently active. Please call endPoll first');
+
+            return;
+        }
+
+        logger.log(`Poll Start: ${poll} ${choices} ${question}`);
+
+        this._doStartPoll(choices, poll, question);
+
+        // inform participants about the new poll.
+        this.conference.sendMessage({
+            type: MESSAGE_POLL_STARTED,
+            event: {
+                choices,
+                poll,
+                question
+            }
+        });
+
+        this._savePollInProsodyModule(roomjid);
+    }
+    /* eslint-enable max-params */
+
+    /**
+     * Vote for a choice in the poll.
+     * @param {string} choiceID - ID of the voted choice
+     * @param {string} userID - ID of user who voted.
+     */
+    voteInPoll(roomjid, choiceID) {
+        if (!choiceID) {
+            return;
+        }
+
+        logger.log(`Voted for ${choiceID}`);
+
+        const myid = this.conference.myUserId();
+
+        this._doVote(choiceID, myid);
+
+        // inform others about my vote.
+        this.conference.sendMessage({
+            type: MESSAGE_POLL_VOTE,
+            event: {
+                choiceID,
+                userID: myid
+            }
+        });
+
+        this._savePollInProsodyModule(roomjid);
+    }
+
+    /**
+     * End the current poll.
+     */
+    endPoll(roomjid) {
+        if (!this.poll && !this.choices && !this.question) {
+            logger.warn('Attempt to end a non existing poll.');
+
+            return;
+        }
+
+        this._doEndPoll();
+
+        // inform others, that I ended the poll
+        this.conference.sendMessage({
+            type: MESSAGE_POLL_END
+        });
+
+        this._savePollInProsodyModule(roomjid);
+    }
+
+    /**
+     * Emit end poll event.
+     */
+    _doEndPoll() {
+        logger.log('Poll has ended');
+
+        this.choices = null;
+        this.poll = null;
+        this.question = null;
+
+        this.conference.eventEmitter.emit(
+            JitsiConferenceEvents.POLL_FINISHED
+        );
+    }
+
+    /**
+     * Emit poll start event.
+     * @param {Object} choices - Choices object.
+     * @param {Object} poll - Poll object.
+     * @param {Object} question - Quesiotn object.
+     */
+    _doStartPoll(choices, poll, question) {
+
+        this.choices = choices;
+        this.poll = poll;
+        this.question = question;
+
+        this.conference.eventEmitter.emit(
+            JitsiConferenceEvents.POLL_STARTED,
+            choices,
+            poll,
+            question
+        );
+    }
+
+    /**
+     * Emit user voted event.
+     * @param {string} choiceID - Choice ID.
+     * @param {string} userID - ID of user who voted.
+     */
+    _doVote(choiceID, userID) {
```

Comment:
Maybe this should be a toggleVote? I don't feel strongly about it though.

Reply:
Yeah, its better